src/index.ts

```diff
@@ -55,25 +50,19 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
     sdkName,
     sdkVersion,
   });
-  const configurationRequestor = new ExperimentConfigurationRequestor(
-    configurationStore,
-    httpClient,
-  );
-  clientInstance = new EppoClient(configurationRequestor, config.assignmentLogger);
-  if (!configurationStore.isInitialized()) {
+  const configurationRequestor = new ExperimentConfigurationRequestor(localStorage, httpClient);
+  if (sessionStorage.get(SESSION_ASSIGNMENT_CONFIG_LOADED) !== 'true') {
     await configurationRequestor.fetchAndStoreConfigurations();
+    sessionStorage.set(SESSION_ASSIGNMENT_CONFIG_LOADED, 'true');
   }
-  return clientInstance;
 }
 
 /**
  * Used to access a singleton SDK client instance.
  * Use the method after calling init() to initialize the client.
+ * @param assignmentLogger a logging implementation to send variation assignments to a data warehouse.
```

Comment:
Ya, that's true. It could work if the `init` method were synchronous (just kick off the request to fetch configs, don't wait for a response). Although that makes error reporting and testing more difficult.

Reply:
Or we just accept that logging can't happen until initialization completes

Reply:
Actually I think it can be solved by defining a static instance on the EppoClient class